testdata/kubecfg_test.jsonnet

```diff
@@ -194,6 +194,18 @@ local result =
   local two = kubecfg.layouts.gvkNsName(one, obj('b'));
   std.assertEqual(two, { 'example.com/v1alpha1.Test': { _: { a: obj('a'), b: obj('b') } } }) &&
 
+  local nested_obj = {
+      foo:: {
+          bar:: {
+              baz:: 'nested!'
+          }
+      }
+  };
+  std.assertEqual(kubecfg.getAtPath(nested_obj, 'foo.bar.baz', 'default!'), 'nested!') &&
```

Comment:
how does it fail when we try something like:

```
kubecfg.hasAtPath({foo:10), 'foo.bar.baz'), true)
```

Reply:
Ah yeah, I see what you mean. It assumes that the value is nested as an object at all times

```jsonnet
  std.assertEqual(kubecfg.hasAtPath({foo: 10}, 'foo.bar.baz'), false) && ...
```

Leads to
```
go run main.go show testdata/kubecfg_test.jsonnet --alpha
ERROR error reading testdata/kubecfg_test.jsonnet: RUNTIME ERROR: Unexpected type number, expected object
```

Reply:
@mkmik Thoughts on [this addition](https://github.com/kubecfg/kubecfg/pull/342/commits/cc5e89a46afa11a35cfcdd1631593305bd33e138) to this?

I've added it as a work in progress commit so it can be reverted if so. It makes sense to me, but I'm also not sure whether the behaviour would be what we would desire.

I.e. `{foo: 10}` and trying to find `foo.bar.baz` _should_ be `false` because that path doesn't exist, compared to erroring out. Simply adding the `std.type` check for the edge case of passing in a small object like :point_up: 

Perhaps there's an edge case I'm missing, but I think either way works fine, i.e. get the `default` or error on the "semi-correct" path where the field is a `number`.

Reply:
to be fair, I don't know what it should be, I was just asking what would happen.

I think it's it's ok if it throws an error.

The default value is meant to to be useful when the data is missing, not when the data has the wrong shape.

Reply:
> I think it's it's ok if it throws an error.

Agreed, I was thinking that using the wrong shape can also be interpreted as "missing", but I think erroring is maybe better in this situation too since it can be an issue on the caller's side.

Either works for sure, I can change it back. Your call on whichever you would prefer in the project :smile:

Reply:
yeah I think erroring is better

Reply:
Reverted to the previous state :+1: